### ISSUE TITLE
Task :hex_fetcher erroring with CaseClauseError

### ISSUE THREAD

**Issue by sorentwo** [hexpm/hex#821]
* Erlang/OTP: 22
* Elixir: 1.10.1
* Hex: 0.20.6-dev

---

As part of a recent experiment I upgraded my local hex to `0.20.6-dev`. Today, running `mix hex.update oban_web` caused a `CaseClauseError`. Note that `oban_web` is a private hex package, if that makes any difference. Here is a sample of the output, including the stacktrace:

```
Upgraded:
  db_connection 2.2.2 => 2.3.0
  oban_web 2.2.2 => 2.2.3
  postgrex 0.15.6 => 0.15.7

12:08:20.599 [error] Task :hex_fetcher started from #PID<0.445.0> terminating
** (CaseClauseError) no case clause matching: {:ok, <<7, 220, 130, 29, 120, 210, 159, 13, 43, 249, 48, 81, 212, 33, 215, 130, 210, 158, 31, 178, 21, 108, 57, 180, 213, 118, 72, 95, 19, 1, 60, 6>>}
    (hex 0.20.6-dev) lib/hex/scm.ex:400: Hex.SCM.fetch/3
    (elixir 1.10.1) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
    (elixir 1.10.1) lib/task/supervised.ex:35: Task.Supervised.reply/5
    (stdlib 3.11) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Function: #Function<7.56784114/0 in Hex.SCM.prefetch/1>
    Args: []
* Updating oban_web (Hex package)
```

The stacktrace points to `hex/scm.ex:400`, which was changed in #812 (1ca1f95ccc). I can confirm that reverting to `0.20.5` fixed the issue.

**Comment by wojtekmach** [comment]
Thanks for the report, looking at the code this seems to be related to checksums. Did you happen to re-publish oban_web recently by any chance? `rm ~/.hex/cache.ets` should help. We should definitely print a better error message though.

**Comment by sorentwo** [comment]
@wojtekmach Good guess! I published 2.2.3 twice within a couple of minutes because I found an issue with the initial release. I may have pulled the initial version from hex between those two versions, but I can't say for sure.
